src/octopeer-github/main/Options/Options.ts

```diff
@@ -136,9 +136,26 @@ const Options = new (class Options {
      */
     public get(optionName: string) {
         if (optionName in this.optionMap) {
-            return this.optionMap[optionName];
+            return this.optionMap[optionName] && this.dependentOption(optionName);
         }
         return false;
     }
 
+    /**
+     * See if any options dependent on the given option are disabled,
+     * If any of the dependent on options is disabled than return false.
+     * If the option is allowed return true.
+     * @param optionName the option to check
+     * @returns {boolean}
+     */
+    private dependentOption(optionName: string): boolean {
+        let dependencies: { [name: string]: string; } = {
+            dataHTML: "dataKeystrokes",
+            dataKeystrokes: "dataComments",
+            mouseClick: "mouseHover",
+            mouseHover: "mousePosition",
+        };
+
+        return !dependencies.hasOwnProperty(optionName) || this.get(dependencies[optionName]);
```

Comment:
Let's do a quick check by constructing a graph from the "map" (javascript objects are basically maps. Talking about `dependencies` here)
If there will be cycles in this graph, it means there will be an infinite loop.
The graph will look as follows:
`dataHTML -> dataKeystrokes -> dataComments`
`mouseClick -> mouseHover -> mousePosition`
Since there are clearly no cycles in this graph, there is no infinite recursion.